### PR TITLE
Feature/laravel 8 upgrade

### PR DESCRIPTION
**Pull request by bendeeson**
Updates to core to support laravel 8.

### CODE REVIEW

File: src/ConversationBuilder/Conversation.php [hunk 1]
```diff
@@ -741,4 +743,16 @@ public static function conversationWithHistory($id): ?Conversation
...
+
+    /**
+     * Prepare a date for array / JSON serialization.
+     *
+     * @param  \DateTimeInterface  $date
+     * @return string
+     */
+    protected function serializeDate(DateTimeInterface $date)
+    {
+        return $date->format('Y-m-d H:i:s');
```
Comment (bendeeson) [inline src/ConversationBuilder/Conversation.php:756]: See https://laravel.com/docs/7.x/upgrade#date-serialization

File: Tests/Unit/MessageMarkUpGeneratorTest.php [hunk 2]
```diff
@@ -20,25 +20,25 @@ public function testButtonMarkUpGenerator()
         $generator->addButtonMessage('test button', $buttons);
 
         $markUp = $generator->getMarkUp();
-        $this->assertRegexp('/<message disable_text="true" hide_avatar="true">/', $markUp);
-        $this->assertRegexp('/<button-message>/', $markUp);
-        $this->assertRegexp('/<button>/', $markUp);
-        $this->assertRegexp('/<text>Button Text<\/text>/', $markUp);
-        $this->assertRegexp('/<value>Value<\/value>/', $markUp);
-        $this->assertRegexp('/<callback>callback<\/callback>/', $markUp);
+        $this->assertMatchesRegularExpression('/<message disable_text="true" hide_avatar="true">/', $markUp);
```
Comment (bendeeson) [inline Tests/Unit/MessageMarkUpGeneratorTest.php:23]: `assertRegexp` is deprecated in phpunit 9. Replaced by `assertMatchesRegularExpression`

File: Tests/TestCase.php [hunk 3]
```diff
@@ -346,7 +347,8 @@ protected function noMatchConversation()
      */
     private function checkRequirements()
     {
-        $annotations = $this->getAnnotations();
+
+        $annotations = Test::parseTestMethodAnnotations(static::class, $this->getName());
```
Comment (bendeeson) [inline Tests/TestCase.php:351]: `$this->getAnnotations()` removed in phpunit 9. The `Test` util class is used instead.

File: composer.json [hunk 4]
```diff
@@ -18,13 +18,13 @@
     "preferred-install": "source"
   },
   "require-dev": {
-    "phpunit/phpunit": "8.3.5",
+    "phpunit/phpunit": "^9.0",
```
Comment (stu-greenshoots) [inline composer.json:21]: Wondered if we actually needed to specify this or we could rely on what orchestra testbench pulled in. but looking at their composer:

```"phpunit/phpunit": "^8.4 || ^9.0"```

so I guess this lets us specify we want 9 👍

**Review by stu-greenshoots — APPROVED** [review]
Thanks Ben, this looks to have got everything given that the test suite is fully passing.

I'm happy to merge this, unless there is anything you wanted to check together with the DGraph upgrade branch?

**Pull request merged**